backend/src/services/graph_access/graph_access.py

```diff
@@ -20,18 +20,26 @@ async def _request(self, url: str) -> httpx.Response:
             )
             return response
 
-    async def get_user_profile_photo(self) -> str | None:
-        print("entering get_user_profile_photo")
-        response = await self._request("https://graph.microsoft.com/v1.0/me/photo/$value")
+    async def get_user_profile_photo(self, user_id: str) -> str | None:
+        request_url = f"https://graph.microsoft.com/v1.0/me/photo/$value"
+
+        if user_id != "me":
+            request_url = f"https://graph.microsoft.com/v1.0/users/{user_id}/photo/$value"
```

Comment:
```suggestion
        request_url = f"https://graph.microsoft.com/v1.0/me/photo/$value" if user_id == "me" else f"https://graph.microsoft.com/v1.0/users/{user_id}/photo/$value"
```
Subjective, I'm fine as is as well.